File: frontend/js/intents.js
/* ════════════════════════════════════════════
   intents.js — Source unique des intentions du chat
   ════════════════════════════════════════════
   Liste plate des intentions reconnues par le moteur ("UNKNOWN" n'y
   figure pas : c'est toujours la valeur de repli implicite quand rien
   ne correspond, jamais une intention "détectée").

   Cette page (chat.js) définit le détail par intention (regex par
   langue, dans INTENTS) ; ce fichier n'existe que pour la LISTE DES
   NOMS, lue aussi par le backend (backend/app/utils/intents.py, qui
   parse ce fichier tel quel — voir ce module) pour valider la réponse
   d'un provider LLM. Ajouter une intention ici + dans INTENTS (chat.js)
   suffit : le backend ne peut plus être en retard, il relit ce fichier
   à chaque démarrage — aucune synchronisation manuelle côté serveur.
   ════════════════════════════════════════════ */
const KNOWN_INTENTS = [
  'CANCEL_TRIP', 'STATUS', 'GREET', 'CONFIRM', 'CANCEL',
  'MODIFY', 'REQUEST_TRANSPORT', 'HELP', 'HISTORY', 'MAP',
];

File: backend/app/utils/intents.py
"""Source unique des intentions connues — lue directement depuis
frontend/js/intents.js (voir ce fichier) plutôt que dupliquée à la main
ici. Un intent ajouté côté chat.js ne peut donc plus être oublié côté
backend : ce module relit le même fichier à chaque démarrage du process.

Le fichier JS est volontairement un format minimal et prévisible
(`const KNOWN_INTENTS = [...]`) — pas besoin d'exécuter du JS pour
l'extraire, une expression régulière + json.loads suffit et reste
beaucoup plus simple qu'ajouter un outil de build au projet.
"""
import json
import os
import re

_INTENTS_JS_PATH = os.path.abspath(os.path.join(
    os.path.dirname(__file__), "..", "..", "..", "frontend", "js", "intents.js"
))


def _load_known_intents() -> tuple[str, ...]:
    with open(_INTENTS_JS_PATH, encoding="utf-8") as f:
        content = f.read()
    match = re.search(r"KNOWN_INTENTS\s*=\s*(\[[^\]]*\])", content, re.DOTALL)
    if not match:
        raise RuntimeError(f"Impossible de lire KNOWN_INTENTS depuis {_INTENTS_JS_PATH}")
    # JS autorise une virgule finale avant "]", pas JSON — on la retire.
    array_literal = re.sub(r",\s*\]", "]", match.group(1))
    names = json.loads(array_literal.replace("'", '"'))
    # "UNKNOWN" n'est jamais listé dans intents.js (ce n'est pas une
    # intention "détectée") mais reste une valeur de repli valide ici.
    return tuple(names) + ("UNKNOWN",)


KNOWN_INTENTS = _load_known_intents()
